examples/anthos-service-mesh-multicluster/README.md

```diff
@@ -0,0 +1,167 @@
+
+- [Installing ASM using Anthos CLI](https://cloud.google.com/service-mesh/docs/gke-anthos-cli-existing-cluster)
+- [Installing ASM using IstioCtl](https://cloud.google.com/service-mesh/docs/gke-install-existing-cluster)
+- [Adding clusters to an Athos Service Mesh](https://cloud.google.com/service-mesh/docs/gke-install-multi-cluster)
+
+## Description
+
+In [Adding clusters to an Athos Service Mesh](https://cloud.google.com/service-mesh/docs/gke-install-multi-cluster), it shows how to federate service meshes of two Anthos **public** clusters. However, it misses a key instruction to open firewall for the service port to remote cluster. So, you final test of HelloWorld might not work. 
+
+This sample builds on topic of Google's Anthos Service Mesh official installation documents, and adds instruction on how to federate two private clusters, which is more demanded in real customer environments. 
+
+As illustrated in the diagram below, we will create a VPC with three subnets. Two subnets are for private clusters, and one for GCE servers. So, we illustrate using bastion server to access private clusters as in real environment. 
+
+![NetworkImage](./asm-private-multiclusters-intranet.png)
+
+The clusters are not accessible from external network. User can only log into the bastion server via IAP tunnel to gain access to this VPC. A firewall is built to allow IAP tunnel into GCE subnet (Subnet C) only. For bastion server in Subnet C to access Kubernetes APIs of both private clusters, Subnet C CIDR range is added to the "_Master Auth Network_" of both clusters. This is illustrated as blue lines and yellow underscore lines in the diagram above.
+
+Also, in order for both clusters to access the service mesh (Istiod) and service deployed on the other cluster, we need to do the following:
+- The pod CIDR range of one cluster must be added to the "_Master Auth Network_" of the other cluster. This enables one cluster to ping _istiod_ on the other cluster. 
+- A firewall need to be open for one cluster's pod CIDR to access the service port on the other cluster. In this sample, it is port 5000 used by the HelloWord testing application. Because the invocation of service is bidirection in HelloWorld testing application, we will add firewall rule for each direction. 
+
+The infrastruction used in this sample is coded in Terraform scripts. The ASM installation steps are coded in Shell script.     
+
+## Prerequisites
+
+As mentioned in [this document](https://cloud.google.com/service-mesh/docs/gke-install-multi-cluster), there are several prerequisites. 
+
+This guide assumes that you have:
+
+- [A Cloud project](https://cloud.google.com/resource-manager/docs/creating-managing-projects).
+- [A Cloud Billing account](https://cloud.google.com/billing/docs/how-to/manage-billing-account).
+
+Also, multi-cluster configuration has these requirements for the clusters in it:
+
+- All clusters must be on the same network. 
```

Comment:
Are peered VPCs not supported?

Reply:
Sadly no :( [Single networks only](https://cloud.google.com/service-mesh/docs/supported-features#platform_environment)